compiler/sem.nim

```diff
@@ -105,8 +105,8 @@ proc commonType*(x, y: PType): PType =
   var a = skipTypes(x, {tyGenericInst, tyAlias})
   var b = skipTypes(y, {tyGenericInst, tyAlias})
   result = x
-  if b.kind in {tyExpr, tyNil, tyVoid}: result = x
-  elif a.kind in {tyExpr, tyNil}: result = y
+  if a.kind in {tyExpr, tyNil}: result = y
```

Comment:
Why swap these two tests?

Reply:
in the latest version, they are no longer swapped. In first version, I was using tyVoid tag on raise statement and needed to check b for tyVoid before a. In latest version, raise statement has no type as originally, I am checking for node kinds instead, a and b checks are reverted to the original order